[Streams] add mentions on channel going live

palmtree5 opened the pull request:
adds in the possibility for everyone/here mentions on a channel going live for any channels set up via the streamalert command. New command is `streamalert mention <everyone/here/off>`

See also: https://github.com/Twentysix26/Red-DiscordBot/projects/2

Commit:
[Streams] add mentions on channel going live for channels set up with streamalert

Commit:
[Streams] make that command work for hitbox and beam streams too

Commit:
[Streams] fix so new bit for settings is only added if not there

MattiSony commented:
Seems like it doesn't work if there's no mention method setup.(I'm guessing that's what the error is saying)

```
future: <Task finished coro=<Streams.stream_checker() done, defined at D:\VPS Backup\Red-DiscordBot\cogs\streams2.py:353> exception=KeyError('81753183336009728',)>
Traceback (most recent call last):
  File "C:\Users\MattiSony\AppData\Local\Programs\Python\Python35\lib\asyncio\tasks.py", line 239, in _step
    result = coro.send(None)
  File "D:\VPS Backup\Red-DiscordBot\cogs\streams2.py", line 371, in stream_checker
    mention_type = self.settings[channel_obj.server.id]["MENTION_TYPE"]
KeyError: '81753183336009728'
```

Should check if there's a method set and if there isn't default it to off.

palmtree5 commented:
@MattiSony have you made other edits to the streams cog because the code at the line number in your error message doesn't match up with the code at that same line number in https://github.com/palmtree5/Red-DiscordBot/blob/8b317f602f8502c99ecf6aeb9586cde265d41158/cogs/streams.py#L371

EDIT: Actually, I see how that could be an issue now

MattiSony commented:
Copied it straight from here, didn't change anything. Also line number matches if you look a bit more down the error :P

palmtree5 committed:
[Streams] fix for missing server id

Should now still work even if the mention type has never been set (thus the server id isn't in settings)

palmtree5 commented:
Just fixed it anyway by handling the exception

palmtree5 commented:
bump

palmtree5 commented:
for testing, try doing the following;

`streamalert mention everyone`
`streamalert twitch food`

(since food is always up, it's easy to test with that)

palmtree5 commented:
bumping this

MattiSony commented:
Been using it for a month now and haven't had any issues with it.

I do think the handling for when the server haven't set a mention type should be done differently though, instead of printing 'Mention type not set for X' in console make the setting just default to off.

palmtree5 commented:
bump

Twentysix26 commented:
https://github.com/Twentysix26/Red-DiscordBot/commit/b69b5b30056c5b4fb213b20ee0970f22de5c0503
